python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1122,7 +1127,24 @@ def _extractIngestInfo(
     def _prepIngest(self, *datasets: FileDataset, transfer: str | None = None) -> _IngestPrepData:
         # Docstring inherited from Datastore._prepIngest.
         filtered = []
-        for dataset in datasets:
+
+        # Ingest could be given tens of thousands of files. It is not efficient
+        # to check for the existence of every single file (especially if they
+        # are remote URIs) but in some transfer modes the files will be checked
+        # anyhow when they are relocated. For direct or None transfer modes
+        # it is possible to not know if the file is accessible at all.
+        # Therefore limit number of files that will be checked (but always
+        # include the first one).
```

Comment:
Could there be a use case when clients want to check every file even if there are thousands of them? 200 looks like an arbitrary number, maybe we want a new option for that (or per-datastore config parameter)?

Reply:
Yes. It's an issue. It's why part of the conversation was considering always doing the checks for POSIX and only skipping for remote. Gregory's original 150k import example was so long (many hours) that it seemed like it wouldn't ever be something people wanted to do. I picked 200 because that's about the size of the focal plane... and it doesn't take very long.